torch/testing/_internal/inputgen/variable/space.py

```diff
@@ -0,0 +1,233 @@
+import math
+from typing import Any, List, Optional, Union
+
+from torch.testing._internal.inputgen.variable.constants import INT64_MAX, INT64_MIN
+from torch.testing._internal.inputgen.variable.type import (
+    check_vtype,
+    is_integer,
+    ScalarDtype,
+    SUPPORTED_TENSOR_DTYPES,
+    VariableType,
+)
+
+
+class Discrete:
+    def __init__(self, values: Optional[List[Any]] = None):
+        if values is None:
+            self.initialized = False
```

Comment:
IMO we shouldn't accept optional values, instead we should check if `values` has value, if not fail the constructor. I would prefer us to fail early, instead of fail at those API calls.

Reply:
It won't fail if values is None. I actually need values to be optional.
Discrete() = Discrete(None) which is different than Discrete([])
Discrete(None) is an un-initialized discrete set
Discrete([]) is an empty set